var/spack/repos/builtin/packages/genie/package.py

```diff
@@ -0,0 +1,265 @@
+
+    phases = ["configure", "build", "install"]
+
+    def setup_build_environment(self, env):
+        env.set("GENIE", self.stage.source_path)
+        return super(Genie, self).setup_build_environment(env)
+
+    def setup_run_environment(self, env):
+        env.set("GENIE", self.prefix)
+        return super(Genie, self).setup_run_environment(env)
+
+    def configure(self, spec, prefix):
+        configure = Executable("./configure")
+        args = self._configure_args(spec, prefix)
+        configure(*args)
+
+    def build(self, spec, prefix):
+        # parallel build is not supported on GENIE 2
+        self._make(parallel=spec.satisfies("@3:"))
+
+    def install(self, spec, prefix):
+        # GENIE make install does not support parallel jobs
+        self._make("install", parallel=False)
+        # GENIE requires these files to be present at runtime, but doesn"t install them
+        # so we must install them ourselves
+        # install_tree function is injected into scope by spack build_environment.py
+        install_tree("config", os.sep.join((prefix, "config")))
+        install_tree("data", os.sep.join((prefix, "data")))
+
+    def _configure_args(self, spec, prefix):
+        args = [
+            "--prefix=%s" % prefix,
+            "--with-compiler=%s" % os.environ["CC"],
+            "--with-libxml2-inc=%s%slibxml2" % (spec["libxml2"].prefix.include, os.sep),
+            "--with-libxml2-lib=%s" % spec["libxml2"].prefix.lib,
+            "--with-log4cpp-inc=%s" % spec["log4cpp"].prefix.include,
+            "--with-log4cpp-lib=%s" % spec["log4cpp"].prefix.lib,
+            "--with-pythia6-lib=%s" % spec["pythia6"].prefix.lib,
+        ]
+        if self.spec.satisfies("@:2"):
+            args += [
+                "--enable-lhapdf",
+                "--with-lhapdf-inc=%s" % spec["lhapdf5"].prefix.include,
+                "--with-lhapdf-lib=%s" % spec["lhapdf5"].prefix.lib,
+                # must be enabled or some GENIE 2 versions fail to link
+                # this option was removed in GENIE 3
+                "--enable-rwght",
+            ]
+        else:
+            args += [
+                "--enable-lhapdf6",
+                "--with-lhapdf6-inc=%s" % spec["lhapdf"].prefix.include,
+                "--with-lhapdf6-lib=%s" % spec["lhapdf"].prefix.lib,
+            ]
+        if "+vleextension" in self.spec:
+            args += ["--enable-vle-extension"]
+        if "+validationtools" in self.spec:
+            args += ["--enable-validation-tools"]
+        if "+test" in self.spec:
+            args += ["--enable-test"]
+        if "+t2k" in self.spec:
+            args += ["--enable-t2k"]
+        if "+fnal" in self.spec:
+            args += ["--enable-fnal"]
+        if "+atmo" in self.spec:
+            args += ["--enable-atmo"]
+        if "+nucleondecay" in self.spec:
+            args += ["--enable-nucleon-decay"]
+        if "+masterclass" in self.spec:
+            args += ["--enable-masterclass"]
```

Comment:
Are all of these off by default?

Reply:
Yes, there are off by default according to the documentation at: <https://hep.ph.liv.ac.uk/~costasa/genie/get_started.html>.